hstrat/serialization/_col_from_records.py

```diff
@@ -0,0 +1,62 @@
+import itertools as it
+import logging
+import typing
+
+from .._auxiliary_lib import get_hstrat_version, log_once_in_a_row
+from ..genome_instrumentation import (
+    HereditaryStratigraphicColumn,
+    HereditaryStratum,
+    HereditaryStratumOrderedStoreList,
+)
+from ._unpack_differentiae import unpack_differentiae
+
+
+def col_from_records(records: typing.Dict) -> HereditaryStratigraphicColumn:
+
+    if "deposition_ranks" in records or "stratum_annotations" in records:
+        raise NotImplementedError
+
+    if get_hstrat_version() != records["hstrat_version"]:
+        log_once_in_a_row(
+            f"""col_from_records version mismatch, record {
+                records['hstrat_version']
+            } and software {
+                get_hstrat_version()
+            }"""
+        )
+
+    def load_stratum_ordered_store() -> HereditaryStratumOrderedStoreList:
+        store = HereditaryStratumOrderedStoreList()
+
+        for differentia, deposition_rank, annotation in zip(
+            unpack_differentiae(
+                records["differentiae"],
+                differentia_bit_width=records["differentia_bit_width"],
+            ),
+            records.get("stratum_deposition_ranks", it.repeat(None)),
+            records.get("stratum_annotation", it.repeat(None)),
+        ):
+            store.DepositStratum(
+                rank=-1,
```

Comment:
Can you use `deposition_rank` instead off `-1` and change the type annotation on `HereditaryStratumOrderedStoreList` to accept `None` or `int` for this argument?

Reply:
Do we need `rank`? Looks like it's not being used in `DepositStratum()` at all

Reply:
Yeah, make `rank` optional in `HereditaryStratumOrderedStoreList`

Reply:
Yup, done in 21d5e4e